build(deps): update dependencies and freeze with uv (#2003)
canonical/charmcraft: CONTRIBUTING.md

## Development environment

> It's recommended to use `uv` for setting up your development environment, but
> this is not required. `uv` is available [as a snap](https://snapcraft.io/astral-uv)
> and from [official sources](https://docs.astral.sh/uv/getting-started/installation/).

Reviewer: ```suggestion
We recommend uv for setting up your local development environment:

- [uv snap](https://snapcraft.io/astral-uv)
- [Official uv binary](https://docs.astral.sh/uv/getting-started/installation/)
```
